core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -385,11 +385,44 @@ private void initHMC() {
       metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER,
           "io.snappydata.jdbc.EmbeddedDriver");
 
-      try {
-        HiveMetaStoreClient hmc = new HiveMetaStoreClient(metadataConf);
-        SnappyHiveCatalog.this.hmClients.set(hmc);
-      } catch (MetaException me) {
-        throw new IllegalStateException(me);
+      final short numRetries = 3;
+      short count = 0;
+      while (true) {
+        try {
+          HiveMetaStoreClient hmc = new HiveMetaStoreClient(metadataConf);
+          SnappyHiveCatalog.this.hmClients.set(hmc);
+          return;
+        } catch (Exception ex) {
+          Throwable t = ex;
+          boolean noDataStoreFound = false;
+          while (t != null && !noDataStoreFound) {
+            noDataStoreFound = (t instanceof SQLException) &&
+                SQLState.NO_DATASTORE_FOUND.startsWith(
+                    ((SQLException)t).getSQLState());
+            t = t.getCause();
+          }
+          // wait for some time and retry if no data store found error
+          // is thrown due to region not being initialized
+          if (count < numRetries && noDataStoreFound) {
+            try {
+              Misc.getI18NLogWriter().warning(LocalizedStrings.DEBUG,
```

Comment:
@sumwale
Tried using following code but this doesn't log anything in the logs (even at warn and error levels). @hbhanawat  is looking into that part as this appears to be after effect SNAP-1420. Will change the code to use slf4j once that is resolved.
```
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
...
...
logger = LoggerFactory.getLogger(SnappyHiveCatalog.class);
...
...
 logger.error("SnappyHiveCatalog.HMSQuery.initHMC: No datastore found " +
                  "while initializing Hive metastore client. " +
                  "Will retry initialization after 500 milliseconds. " +
                  "Exception received is " + t);
         if (logger.isDebugEnabled()) {
            logger.error("Exception stacktrace:", ex);
         }